src/Toolkit/Toolkit/UI/Controls/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -67,33 +68,40 @@ private void OnSymbolChanged(Symbology.Symbol? oldValue, Symbology.Symbol? newVa
                 {
                     OnEventAction = (instance, source, eventArgs) =>
                     {
-                        Refresh();
+                        _ = Refresh();
                     },
                     OnDetachAction = (instance, weakEventListener) => instance.PropertyChanged -= weakEventListener.OnEvent,
                 };
                 newValue.PropertyChanged += _inpcListener.OnEvent;
             }
 
-            Refresh();
+            _ = Refresh();
         }
 
-        private async void Refresh()
+        private async Task Refresh()
         {
-            if (_currentUpdateTask != null)
+            try
             {
-                // Instead of refreshing immediately when a refresh is already in progress, avoid updating too frequently, but just flag it dirty
-                // This avoid multiple refreshes where properties change very frequently, but just the latest state gets refreshed.
-                _isRefreshRequired = true;
-                return;
-            }
+                if (_currentUpdateTask != null)
+                {
+                    // Instead of refreshing immediately when a refresh is already in progress, avoid updating too frequently, but just flag it dirty
+                    // This avoid multiple refreshes where properties change very frequently, but just the latest state gets refreshed.
+                    _isRefreshRequired = true;
+                    return;
+                }
 
-            _isRefreshRequired = false;
-            var task = _currentUpdateTask = UpdateSwatchAsync();
-            await task;
-            _currentUpdateTask = null;
-            if (_isRefreshRequired)
+                _isRefreshRequired = false;
+                var task = _currentUpdateTask = UpdateSwatchAsync();
+                await task;
+                _currentUpdateTask = null;
+                if (_isRefreshRequired)
+                {
+                    await Refresh();
+                }
```

Comment:
Alternative solution -- functionally same but avoids recursion:
```suggestion
                do
                {
                    _isRefreshRequired = false;
                    var task = _currentUpdateTask = UpdateSwatchAsync();
                    await task;
                } while (_isRefreshRequired);
                _currentUpdateTask = null;
```